src/api/app/controllers/webui/users_controller.rb

```diff
@@ -68,7 +68,9 @@ def create
   end
 
   def update
-    if !User.admin_session? && (User.session! != @displayed_user || !@configuration.accounts_editable?(@displayed_user))
+    if !User.admin_session? &&
+       (User.session! != @displayed_user || !@configuration.accounts_editable?(@displayed_user)) &&
+       !(User.session.is_moderator? && params[:user][:blocked_from_commenting].present?)
```

Comment:
Shouldn't this be handled in a policy?

Reply:
It sounds like a pundit policy, yes.

There is code for updating in the user policy: https://github.com/openSUSE/open-build-service/blob/89a89730ed1770996ff86e911374a1dbdb92b790/src/api/app/policies/user_policy.rb#L2-L4

This code seems to be not in sync with the update actions with the API and Webui controllers for a while. I see that refactoring the update actions to use the same update policy is a refactoring opportunity, but I don't see it necessary belonging to this pull request. Do you mind leaving this refactoring outside this pull request?

Reply:
> Do you mind leaving this refactoring outside this pull request?

I don't mind :+1: